pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/SegmentCompletionManager.java

```diff
@@ -1068,16 +1069,28 @@ private int numReplicasToLookFor() {
       _state = State.COMMITTING;
       // In case of splitCommit, the segment is uploaded to a unique file name indicated by segmentLocation,
       // so we need to move the segment file to its permanent location first before committing the metadata.
-      if (isSplitCommit) {
+      // The committingSegmentDescriptor is then updated with the permanent segment location to be saved in metadata
+      // store.
+      // The exception is when the server uses the Peer segment download scheme, in such case, there is no need to
+      // move the segment.
+      if (isSplitCommit && !isPeerSegmentDownloadScheme(committingSegmentDescriptor)) {
```

Comment:
we dont need to check the peer download scheme here, since we check and put the right value while committing metadata in PinotLLCRealtimeSegmentManager. We can always set the segment location. The if condition can be as before

Reply:
got it. thank.  good idea.